testSrc/unit/io/flutter/utils/StdoutJsonParserTest.java

```diff
@@ -58,12 +57,43 @@ public void split_json() throws Exception {
     parser.appendOutput("[{'foo':");
     parser.appendOutput("'bar'}]\n");
     parser.appendOutput("bye\n");
-    parser.flush();
 
     assertArrayEquals(
       "validating parser results",
       new String[]{"hello\n", "there\n", "[{'foo':'bar'}]\n", "bye\n"},
       parser.getAvailableLines().toArray()
     );
   }
+
+  @Test
+  public void deep_nested_json() throws Exception {
```

Comment:
nit: I believe our test names use lowerCamelCase

Reply:
done. for some reason the existing tests in this file used_under_scores_so_i_copied_them :)